test/pact_test.exs

```diff
@@ -68,4 +55,28 @@ defmodule PactTest do
     Pact.put(:string, Integer)
     assert Pact.get("string") == Integer
   end
+
+  test "replace module replaces the given module" do
+    Pact.put("foo", String)
+
+    Pact.replace self, "foo" do
+      def awesome? do
+        true
+      end
```

Comment:
It takes the block as a quoted expression, so I don't think it's worth adding a test since that's expected behavior?